src/prefect/cli/cloud.py

```diff
@@ -150,13 +288,81 @@ async def login(
     Creates a new profile configured to use the specified PREFECT_API_KEY.
     Uses a previously configured profile if it exists.
     """
+    if not app.console.is_interactive and (not key or not workspace_handle):
+        exit_with_error(
+            "When not using an interactive terminal, you must supply a `--key` and `--workspace`."
+        )
+
+    profiles = load_profiles()
+    current_profile = get_settings_context().profile
+
+    if key and PREFECT_API_KEY.value() == key:
+        exit_with_success("This profile is already authenticated with that key.")
+
+    already_logged_in_profiles = []
+    for name, profile in profiles.items():
+        profile_key = profile.settings.get(PREFECT_API_KEY)
+        if (
+            not key
+            and profile_key is not None
+            or (key and profile_key == key)
+            and await check_key_is_valid_for_login(profile_key)
+        ):
+            already_logged_in_profiles.append(name)
+
+    current_profile_is_logged_in = current_profile.name in already_logged_in_profiles
+
+    if current_profile_is_logged_in:
+        app.console.print("It looks like you're already authenticated on this profile.")
+        should_reauth = typer.confirm(
+            "? Would you like to reauthenticate?", default=False
+        )
+        if not should_reauth:
+            app.console.print("Using the existing authentication on this profile.")
+            key = PREFECT_API_KEY.value()
+
+    elif already_logged_in_profiles:
+        app.console.print(
+            "It looks like you're already authenticated with another profile."
+        )
+        if not typer.confirm(
+            "? Would you like to reauthenticate with this profile?", default=False
+        ):
```

Comment:
In testing, selecting "no" here throws an OS connection error:
```
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 61] Connect call failed ('::1', 3000, 0, 0)
```
Not particularly helpful but worth checking if you can repro

Reply:
Like... every time? That's weird. Do you have a full traceback?

Reply:
Yup, every time for me :/ 

```
anyio._backends._asyncio.ExceptionGroup: 2 exceptions were raised in the task group:
----------------------------
Traceback (most recent call last):
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_core/_sockets.py", line 164, in try_connect
    stream = await asynclib.connect_tcp(remote_host, remote_port, local_address)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 1691, in connect_tcp
    await get_running_loop().create_connection(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection
    raise exceptions[0]
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection
    sock = await self._connect_sock(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock
    await self.sock_connect(sock, address)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect
    return await fut
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 61] Connect call failed ('::1', 3000, 0, 0)
----------------------------
Traceback (most recent call last):
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_core/_sockets.py", line 164, in try_connect
    stream = await asynclib.connect_tcp(remote_host, remote_port, local_address)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 1691, in connect_tcp
    await get_running_loop().create_connection(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 1064, in create_connection
    raise exceptions[0]
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 1049, in create_connection
    sock = await self._connect_sock(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 960, in _connect_sock
    await self.sock_connect(sock, address)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/selector_events.py", line 500, in sock_connect
    return await fut
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/selector_events.py", line 535, in _sock_connect_cb
    raise OSError(err, f'Connect call failed {address}')
ConnectionRefusedError: [Errno 61] Connect call failed ('127.0.0.1', 3000)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_exceptions.py", line 8, in map_exceptions
    yield
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/backends/asyncio.py", line 109, in connect_tcp
    stream: anyio.abc.ByteStream = await anyio.connect_tcp(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_core/_sockets.py", line 222, in connect_tcp
    raise OSError("All connection attempts failed") from cause
OSError: All connection attempts failed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
    yield
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_transports/default.py", line 353, in handle_async_request
    resp = await self._pool.handle_async_request(req)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_async/connection_pool.py", line 253, in handle_async_request
    raise exc
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_async/connection_pool.py", line 237, in handle_async_request
    response = await connection.handle_async_request(request)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_async/connection.py", line 86, in handle_async_request
    raise exc
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_async/connection.py", line 63, in handle_async_request
    stream = await self._connect(request)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_async/connection.py", line 111, in _connect
    stream = await self._network_backend.connect_tcp(**kwargs)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/backends/auto.py", line 29, in connect_tcp
    return await self._backend.connect_tcp(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/backends/asyncio.py", line 107, in connect_tcp
    with map_exceptions(exc_map):
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpcore/_exceptions.py", line 12, in map_exceptions
    raise to_exc(exc)
httpcore.ConnectError: All connection attempts failed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/cli/_utilities.py", line 41, in wrapper
    return fn(*args, **kwargs)
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/utilities/asyncutils.py", line 201, in coroutine_wrapper
    return run_async_in_new_loop(async_fn, *args, **kwargs)
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/utilities/asyncutils.py", line 152, in run_async_in_new_loop
    return anyio.run(partial(__fn, *args, **kwargs))
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_core/_eventloop.py", line 70, in run
    return asynclib.run(func, *args, **backend_options)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 292, in run
    return native_run(wrapper(), debug=debug)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 287, in wrapper
    return await func(*args)
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/cli/cloud.py", line 365, in login
    workspaces = await client.read_workspaces()
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/client/cloud.py", line 70, in read_workspaces
    return pydantic.parse_obj_as(List[Workspace], await self.get("/me/workspaces"))
  File "/Users/nicholasbrown/Projects/prefect/src/prefect/client/cloud.py", line 90, in get
    res = await self._client.get(route, **kwargs)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1751, in get
    return await self.request(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1527, in request
    return await self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1614, in send
    response = await self._send_handling_auth(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1642, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1679, in _send_handling_redirects
    response = await self._send_single_request(request)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_client.py", line 1716, in _send_single_request
    response = await transport.handle_async_request(request)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_transports/default.py", line 352, in handle_async_request
    with map_httpcore_exceptions():
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/Users/nicholasbrown/.conda/envs/prefect/lib/python3.10/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ConnectError: All connection attempts failed
An exception occurred.
```

but if you're not able to repro I wouldn't be surprised if this boiled down to a dirty env.